feature/foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/ForYouViewModel.kt

```diff
@@ -68,13 +68,15 @@ class ForYouViewModel @Inject constructor(
                 } else {
                     getSaveableNewsResources(
                         filterTopicIds = userData.followedTopics,
-                    ).mapToFeedState()
+                    )
+                        .map<List<UserNewsResource>, NewsFeedUiState>(NewsFeedUiState::Success)
                 }
             }
             // Flatten the feed flows.
             // As the selected topics and topic state changes, this will cancel the old feed
             // monitoring and start the new one.
             .flatMapLatest { it }
+            .onStart { emit(NewsFeedUiState.Loading) }
```

Comment:
If we still use `onStart`, if we leave the bookmarks screen for > 5 seconds, the bug will show up again. Can we drop  emitting `Loading` each time we resume the stream? It wipes out the cached state in the `StateFlow`.

Reply:
I don't think we need `Async`. The initial value for the news feed is already set to `Loading`. 

The only scenario I can think of where we'd want to show a loading indicator to the user again would be during a network sync, but this is already handled by a separate flow in `isSyncing`.

Reply:
+1, or at least changing the types here so that it is more clear what behavior we want.

Right now, we clear out the old data when we are loading, which ends up leading to behavior that feels weird. The behavior we want is that we can continue showing the old data while we are loading (but we still want to indicate that we are reloading).

I feel like we should allow the `Loading` state to also have a list of items to display, which would be the set of old items while we are loading new ones.

Reply:
@tunjid
StateFlow will not reset cache value to Loading (it will reset if we use `WhileSubscribed(replayExpiration = Duration.ZERO)`.

_Sent from my 2201117TG using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

Reply:
@hoc081098 `onStart` will be called when the flow is restarted, which will happen if the app is backgrounded for >5 seconds and then brought to the foreground. This will cause the flow to emit a `Loading` state.

Reply:
My bad 😂

Reply:
I think we were overthinking this... unless I'm missing something obvious `onStart` isn't needed there. `stateIn`'s `initialValue` should already give us the loading signal while data is fetched for the first time